sdk/core/azure-core/CMakeLists.txt

```diff
@@ -23,13 +23,35 @@ find_package(Threads REQUIRED)
 
 # min version for `CURLSSLOPT_NO_REVOKE`
 # https://curl.haxx.se/libcurl/c/CURLOPT_SSL_OPTIONS.html
-set(CURL_MIN_REQUIRED_VERSION 7.44)
+set(CURL_MIN_REQUIRED_VERSION 7.4)
 find_package(CURL ${CURL_MIN_REQUIRED_VERSION} CONFIG QUIET)
 if(NOT CURL_FOUND)
   find_package(CURL ${CURL_MIN_REQUIRED_VERSION} REQUIRED)
 endif()
 message("Libcurl version ${CURL_VERSION_STRING}")
 
+########################## Libcurl per-version found features. #################################
+#                                                                                              #
+#  Option `CURLSSLOPT_NO_REVOKE` will only be available for customers using version >= 7.44    #
+#                                                                                              #
+#  However, the Azure SDK for C++ can use an older version 7.4 witout supporting this option   #
+#                                                                                              #
+#    #     #              #            #           #          #            #                   #
+string(REPLACE "." ";" CURL_VERSION_LIST ${CURL_VERSION_STRING})
+list(GET CURL_VERSION_LIST 0 CURL_VERSION_MAJOR)
+list(GET CURL_VERSION_LIST 1 CURL_VERSION_MINOR)
+list(GET CURL_VERSION_LIST 2 CURL_VERSION_PATCH)
+set(OPTION_NO_REVOKE_MIN_MAJOR_VERSION "7")
+set(OPTION_NO_REVOKE_MIN_MINOR_VERSION "44")
+if(${CURL_VERSION_MAJOR} GREATER_EQUAL ${OPTION_NO_REVOKE_MIN_MAJOR_VERSION} AND ${CURL_VERSION_MINOR} GREATER_EQUAL ${OPTION_NO_REVOKE_MIN_MINOR_VERSION})
```

Comment:
Rather than doing custom logic like this, can we use the built-in [VERSION_GREATER](https://cmake.org/cmake/help/latest/command/if.html#condition-syntax) check using semver?

Reply:
I believe the current check is wrong. It will consider 8.0 to be not suitable since 0 < 44, even though the major version is higher.

If we want to use our own logic, the feature is supported if:
- curl major version is higher OR
- curl major version is the same, and minor version is >=

So, we need to update the check.